src/qc.rs

```diff
@@ -121,3 +123,53 @@ macro_rules! arbitrary_large {
 
 arbitrary_large!(I64: i64);
 arbitrary_large!(U64: u64);
+
+macro_rules! arbitrary_float {
+    ($id:ident: $ty:ident) => {
+        #[derive(Clone, Copy)]
+        pub struct $id(pub $ty);
+
+        impl Arbitrary for $id {
+            fn arbitrary<G>(g: &mut G) -> $id
+                where G: Gen
+            {
+                let special = [
+                    -0.0, 0.0, 1.0, $ty::NAN, $ty::INFINITY, -$ty::INFINITY
+                ];
+
+                if g.gen() { // Random special case
+                    let index: usize = g.gen();
+                    $id(special[index % special.len()])
+                } else if g.gen() { // Random anything
+                    let sign: bool = g.gen();
+                    let exponent: <$ty as Float>::Int = g.gen();
+                    let significand: <$ty as Float>::Int = g.gen();
+                    $id(<$ty>::from_parts(sign, exponent, significand))
+                } else if g.gen() { // Denormalized
+                    let sign: bool = g.gen();
+                    let exponent: <$ty as Float>::Int = 0;
+                    let significand: <$ty as Float>::Int = g.gen();
+                    $id(<$ty>::from_parts(sign, exponent, significand))
+                } else { // NaN variants
+                    let sign: bool = g.gen();
+                    let exponent: <$ty as Float>::Int = g.gen();
+                    let significand: <$ty as Float>::Int = 0;
+                    $id(<$ty>::from_parts(sign, exponent, significand))
+                }
```

Comment:
I'm not entirely convinced by the above method, but the tests get run often enough that I believe that the different combinations will be run many times, and not having any state keeps the method simple.